### ISSUE TITLE
Finetuning is not progressing, it just stops, remaining time is a question mark.

### ISSUE THREAD

**Issue by szekrenyesi** [vasistalodagala/whisper-finetune#17]
ARGUMENTS OF INTEREST:
{'model_name': 'openai/whisper-large-v3', 'language': 'hungarian', 'sampling_rate': 16000, 'num_proc': 2, 'train_strategy': 'epoch', 'learning_rate': 0.003, 'warmup': 1000, 'train_batchsize': 16, 'eval_batchsize': 8, 'num_epochs': 20, 'num_steps': 100000, 'resume_from_ckpt': 'None', 'output_dir': 'models', 'train_datasets': ['/data/train'], 'eval_datasets': ['/data/dev']}

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
DATASET PREPARATION IN PROGRESS...
Loading cached shuffled indices for dataset at /tmp/tmpiriwoix_/data//train/cache-b8d5b2a7daea0d9d.arrow
Loading cached shuffled indices for dataset at /tmp/tmpiriwoix_/data/dev/cache-fedae2d849cf9c83.arrow
Map (num_proc=2):   0%|                                                                                                                                                     | 0/1000 [00:00<?, ? examples/s]

**Comment by szekrenyesi** [comment]
It is solved. The number of process had to be 1 since I used only one dataset for training

**Comment by szekrenyesi** [comment]
It is

**Comment by J-Korn** [comment]
I am getting the same thing and I see your issue is marked as closed. Did you end up solving it?

 torchrun train/fine-tune_on_hf_dataset.py --model_name openai/whisper-large --language Greek --train_datasets google/fleurs --train_dataset_configs el_gr --train_dataset_splits train --train_dataset_text_columns transcription --eval_datasets google/fleurs --eval_dataset_splits validation --eval_dataset_text_columns transcription --eval_dataset_configs el_gr


+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++


ARGUMENTS OF INTEREST:
{'model_name': 'openai/whisper-large', 'language': 'Greek', 'sampling_rate': 16000, 'num_proc': 2, 'train_strategy': 'steps', 'learning_rate': 1.75e-05, 'warmup': 20000, 'train_batchsize': 48, 'eval_batchsize': 32, 'num_epochs': 20, 'num_steps': 100000, 'resume_from_ckpt': None, 'output_dir': 'output_model_dir', 'train_datasets': ['google/fleurs'], 'train_dataset_configs': ['el_gr'], 'train_dataset_splits': ['train'], 'train_dataset_text_columns': ['transcription'], 'eval_datasets': ['google/fleurs'], 'eval_dataset_configs': ['el_gr'], 'eval_dataset_splits': ['validation'], 'eval_dataset_text_columns': ['transcription']}


+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++


Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
DATASET PREPARATION IN PROGRESS...
/usr/local/lib/python3.10/dist-packages/datasets/load.py:1486: FutureWarning: The repository for google/fleurs contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/google/fleurs
You can avoid this message in future by passing the argument `trust_remote_code=True`.
Passing `trust_remote_code=True` will be mandatory to load this dataset from the next major release of `datasets`.
  warnings.warn(
Map (num_proc=2):   0%|                   | 0/3215 [00:00<?, ? examples/s]

**Comment by seongminp** [comment]
Hi. did you solve this issue? I am stuck at the same place

**Comment by seongminp** [comment]
To anyone stumbling upon this issue, the solution is either
1. --num_proc 1 or
2. torch.set_num_threads(1) ([source](https://discuss.huggingface.co/t/using-num-proc-1-in-dataset-map-hangs/44310/4))
